app/controllers/guest_panel/ContactController.java

```diff
@@ -0,0 +1,100 @@
+package controllers.guest_panel;
+
+import authorization.Authenticator;
+import authorization.UsernameAlreadyExistsException;
+import authorization.models.UserForm;
+import helpers.SessionMessages;
+import models.DatabaseException;
+import models.Message;
+import models.Model;
+import play.data.Form;
+import play.data.FormFactory;
+import play.data.validation.ValidationError;
+import play.db.Database;
+import play.mvc.Controller;
+import play.mvc.Http;
+import play.mvc.Result;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Created by szymon on 5/8/17.
+ */
+public class ContactController  extends Controller {
+    private final play.data.FormFactory formFactory;
+
+    @Inject
+    public ContactController(Database db, FormFactory formFactory) {
+        this.formFactory = formFactory;
+    }
+
+    public Result get() {
+        Form<Message> form = formFactory.form(Message.class);
+        Http.Context context = Http.Context.current();
+        List<MessageWrapper> messageWrapperList =  MessageWrapper.getHistory(Authenticator.getUser().id);
+        return ok(views.html.guest_panel_views.contact.render(form, messageWrapperList, context.messages()));
+    }
+
+    public Result post() {
+        Form<Message> form = formFactory.form(Message.class).bindFromRequest();
+        Http.Context context = Http.Context.current();
+        if(form.hasErrors()) {
+            List<MessageWrapper> messageWrapperList =  MessageWrapper.getHistory(Authenticator.getUser().id);
+            return ok(views.html.guest_panel_views.contact.render(form, messageWrapperList, context.messages()));
+        }
+
+        Message message = form.get();
+
+        if(form.hasErrors()) {
+            List<MessageWrapper> messageWrapperList =  MessageWrapper.getHistory(Authenticator.getUser().id);
+            return ok(views.html.guest_panel_views.contact.render(form, messageWrapperList, context.messages()));
+        }
+        message.date = new Date().getTime();
+        message.sender = Authenticator.getUser().id;
+        message.addressee = Message.getManagerId();
+        message.save();
+        SessionMessages.addSuccess("Your message has been sent!");
+        List<MessageWrapper> messageWrapperList =  MessageWrapper.getHistory(Authenticator.getUser().id);
+        return ok(views.html.guest_panel_views.contact.render(form, messageWrapperList, context.messages()));
```

Comment:
Redirect to get() instead. Otherwise pressing F5 would ask you to resubmit the form.